src/connection/connection_v2.ts

```diff
@@ -70,6 +72,93 @@ export class ConnectionV2 extends BaseConnection {
     return this.engineEndpoint;
   }
 
+  // Async methods
+  async executeAsync(
+    query: string,
+    executeQueryOptions: ExecuteQueryOptions = {}
+  ): Promise<AsyncStatement> {
+    const { httpClient } = this.context;
```

Comment:
Look like we have a lot of code duplication with sync execute method. Can we refactor common logic into separate functions to reuse?